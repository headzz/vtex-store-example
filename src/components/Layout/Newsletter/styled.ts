import { GREY_LVL1, WHITE } from 'configuration/constants'
import styled from 'styled-components'

export const NewsletterStyled = styled.section`
  background: linear-gradient(90deg, ${WHITE} 50%, ${GREY_LVL1} 50%);
  padding: 6.4rem 0;
  width: 100%;
`
