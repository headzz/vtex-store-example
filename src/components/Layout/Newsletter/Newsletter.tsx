import { NewsletterStyled } from './styled'

/**
 * @component
 * @name Newsletter
 * @description Defines the container for the layout of newsletter section.
 * @namespace components components/Layout/Newsletter
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Newsletter: React.FC = ({ children }) => {
  return <NewsletterStyled>{children}</NewsletterStyled>
}

export default Newsletter
