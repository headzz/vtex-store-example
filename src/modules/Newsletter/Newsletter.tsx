import Layout from 'components/Layout'

import TextArea from './components/TextArea'
import { ContactContainer } from './styled'

/**
 * @component
 * @name Newsletter
 * @description creates Newsletter component section to use on screen.
 * @namespace components modules/Newsletter
 *
 */
const Newsletter: React.FC = () => (
  <Layout.Newsletter>
    <ContactContainer>
      <TextArea />
    </ContactContainer>
  </Layout.Newsletter>
)

export default Newsletter
