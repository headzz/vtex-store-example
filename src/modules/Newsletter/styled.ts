import styled from 'styled-components'

export const ContactContainer = styled.div`
  display: flex;
  justify-content: center;
  div {
    padding: 0 1rem;
    width: 100%;

    p:first-child {
      margin-bottom: 3.2rem;
      text-align: center;
    }
  }
  @media (min-width: 768px) {
    div {
      padding: 0;
      width: 67.6rem;
    }
  }
`
