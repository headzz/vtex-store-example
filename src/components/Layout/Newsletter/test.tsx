import { render } from '@testing-library/react'

import Newsletter from './Newsletter'

describe('<Newsletter />', () => {
  it('should render component that wraps newsletter on page', () => {
    const { container } = render(<Newsletter />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
