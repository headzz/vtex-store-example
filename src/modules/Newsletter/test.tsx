import { render } from '@testing-library/react'

import Newsletter from './Newsletter'

describe('Newsletter', () => {
  it('match snapshot', () => {
    const { container } = render(<Newsletter />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
