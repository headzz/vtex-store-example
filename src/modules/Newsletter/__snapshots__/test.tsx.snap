// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter match snapshot 1`] = `
.c0 {
  background-color: #000000;
  padding: 13.6rem 0;
  width: 100%;
}

.c3 {
  color: #03FFA5;
  font-size: 3.2rem;
  font-weight: 700;
}

.c4 {
  color: #ffffff;
  font-size: 3.2rem;
  font-weight: 700;
}

.c8 {
  color: #03FFA5;
  font-size: 1.6rem;
  font-weight: 500;
}

.c7 {
  border: 0;
  border-radius: 0;
  background-color: transparent;
  outline: 0;
  padding: 1.2rem 2rem;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c2 p {
  text-align: center;
}

.c2 p:first-child {
  position: absolute;
  left: -4.8rem;
  top: 0;
}

.c2 p:nth-child(2) {
  margin-bottom: 3.2rem;
}

.c6 {
  background-color: transparent;
  border: 0;
  border-bottom: 1px solid #03FFA5;
  color: #ffffff;
  font-family: 'Barlow';
  outline: 0;
  padding: 0 9rem 1.8rem 0;
  width: 100%;
}

.c5 {
  background-color: transparent;
  position: relative;
  width: 50rem;
}

.c5 button {
  font-family: 'Barlow',sans-serif;
  left: calc(100% - 50px);
  position: absolute;
  top: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 div {
  padding: 0 1rem;
  width: 100%;
}

.c1 div p:first-child {
  margin-bottom: 3.2rem;
  text-align: center;
}

@media (min-width:768px) {

}

@media (min-width:768px) {
  .c1 div {
    padding: 0;
    width: 67.6rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
        color="#03FFA5"
        title="small"
      >
        //
      </p>
      <p
        class="c4"
        color="#ffffff"
        title="small"
      >
        RECEBA NOVIDADES DA NOSSA √ÅREA DE PRODUTOS DIGITAIS.
      </p>
      <div
        class="c5"
      >
        <input
          class="c6"
          type="text"
        />
        <button
          class="c7"
        >
          <p
            class="c8"
            color="#03FFA5"
            title=""
          >
            CADASTRAR
          </p>
        </button>
      </div>
    </div>
  </div>
</section>
`
